### PR TITLE
[CPDEV-101357] Remove PSP and master role support

### LINKED ISSUE #55: Deprecated label for nodes
There are warning messages in dump/degug.log:
```
 Applying label node-role.kubernetes.io/control-plane='' to Nodes with label node-role.kubernetes.io/master='' (deprecated)
```
It is related to [Replace taint master with control-plane](https://github.com/kubernetes/website/pull/29935).
In other words, `node-role.kubernetes.io/master` label will be replaced with `node-role.kubernetes.io/control-plane` label in Kubernetes v1.24

There are several files related to the label in Kubetools:
```
kubetool/procedures/check_paas.py:                        if 'node-role.kubernetes.io/master' not in node_description['metadata']['labels']:
kubetool/templates/plugins/calico-v3.20.yaml.j2:        - key: node-role.kubernetes.io/master
kubetool/templates/plugins/calico-v3.19.yaml.j2:        - key: node-role.kubernetes.io/master
kubetool/kubernetes.py:                node["taints"].append("node-role.kubernetes.io/master:NoSchedule-")
```
It should be decided how to deal with the new label.

### PR DESCRIPTION
**Pull request by ilia1243**
### Description
* Remove Kubernetes v1.23, v1.24 and all related code.
* Remove PSP support and all related code.
* No longer support `master` role.
    * Support of this role was preserved for `do` procedure.

Fixes #55

### Breaking changes
* `master` is not supported neither in `nodes.roles` nor in other sections that refer to roles, e.g `thirdparties.group`.
* Checks of `master` role existence in jinja templates will no longer work, e. g. `{{ nodes | select("has_role", "master") | list }}`

### Checklist
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Integration CI passed
- [x] Unit tests. If Yes list of new/changed tests with brief description
- [x] There is no merge conflicts

#### Unit tests
test_do.py - cover support of `master` role for `do` procedure.

### CODE REVIEW

**Review by koryaga — APPROVED** [review]

**Review by n549 — COMMENTED** [review]
Probably it's worth to remove [Service Account Issuer](https://github.com/Netcracker/KubeMarine/blob/deprecation/remove_psp_support/documentation/Installation.md#service-account-issuer) description from IN? This FeatureGate was [removed in v1.23](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates-removed/).

**Comment by ilia1243** [comment]
> Probably it's worth to remove [Service Account Issuer](https://github.com/Netcracker/KubeMarine/blob/deprecation/remove_psp_support/documentation/Installation.md#service-account-issuer) description from IN? This FeatureGate was [removed in v1.23](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates-removed/).

Only remove the `feature-gates: "ServiceAccountIssuerDiscovery=true"` or the whole description? As far as I see, related flags are still available https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/.

**Comment by ilia1243** [comment]
> > Only remove the `feature-gates: "ServiceAccountIssuerDiscovery=true"` or the whole description? As far as I see, related flags are still available https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/.
> 
> yes, remove the `feature-gates: "ServiceAccountIssuerDiscovery=true"`. And `Service Account Issuer feature is supported only on Kubernetes 1.20.` warning is not needed any more.

It is necessary to repeat the steps and make sure it is still working. Also there are few other guides that require actualization. Created issue #666

**Review by Imadzuma — APPROVED** [review]

File: kubemarine/templates/plugins/iperf3.yaml.j2
Comment (alexarefev) [inline kubemarine/templates/plugins/iperf3.yaml.j2:1]: Do we need that Jinja template?

Comment (ilia1243) [reply kubemarine/templates/plugins/iperf3.yaml.j2:1]: Added missed guide and support PSS for this external plugin in #667

**Pull request merged**